kubernetes/migration/spec.md

```diff
@@ -0,0 +1,33 @@
+# K8s/CoreDNS Corefile Migration Tools
+
+This Go library provides a set of functions to help handle migrations of CoreDNS Corefiles to be compatible with new versions of CoreDNS.  The task of upgrading CoreDNS is the responsibility of a variety of Kubernetes management tools (e.g. kubeadm and others), and the precise behavior may be different for each one.  This library abstracts some basic helper functions that make this easier to implement.
+
+## Proposed functions:
+
+`Deprecated(fromVersion, toVersion, corefile string) []string`: returns a list of deprecated plugins or directives present in the Corefile.
+
+`Removed(fromVersion, toVersion, corefile string) []string`: returns a list of removed plugins or directives present in the Corefile.
+
+`Migrate(fromVersion, toVersion, corefile string, deprecations boolean) (string, error)`: returns an automatically migrated version of the Corefile, or an error if it cannot. It should:
```

Comment:
is the `fromVersion` useful ? 
should we make a validation that the `corefile` is valid for the `fromVersion` ?

Reply:
Yes, fromVersion is required.  It usually won't matter, but it could.